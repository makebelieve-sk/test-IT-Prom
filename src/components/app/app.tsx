import React from 'react';

import { MainBlock } from '../main-block/main-block';

export const App: React.FC = () => {
    return (
        <div className="main-wrapper">
            <MainBlock />
        </div>
    );
};