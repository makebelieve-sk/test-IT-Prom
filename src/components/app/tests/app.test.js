import React from 'react';
import ReactDOM from 'react-dom'; 
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { App } from '../app';

Enzyme.configure({ adapter: new Adapter() });

it(`App correctly rendering`, () => {
    const tree = renderer  
        .create(<App />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});

describe('App correctly rendering and functioning', () => {
    let wrapper;
  
    beforeEach(() => {
        wrapper = shallow(<App />);
    });

    it(`+++ App renders without crashing`, () => {
        const div = document.createElement('div');
      
        ReactDOM.render(<App />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('+++ render the connected App', () => {
        expect(wrapper.length).toEqual(1);
    });
});