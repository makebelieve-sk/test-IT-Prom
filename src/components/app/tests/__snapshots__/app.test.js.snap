// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App correctly rendering 1`] = `
<div
  className="main-wrapper"
>
  <div
    className="main-block"
  >
    <div
      className="list-block"
    >
      <div
        className="button-in-list"
        onClick={[Function]}
      >
        Профессии
      </div>
      <div
        className="button-in-list"
        onClick={[Function]}
      >
        Отделы
      </div>
      <div
        className="button-in-list"
        onClick={[Function]}
      >
        Сотрудники
      </div>
    </div>
  </div>
</div>
`;
